geo-traits/src/coord.rs

```diff
@@ -0,0 +1,141 @@
+use std::marker::PhantomData;
+
+use geo_types::{Coord, CoordNum};
+
+use crate::Dimensions;
+
+/// A trait for accessing data from a generic Coord.
+///
+/// Refer to [geo_types::Coord] for information about semantics and validity.
+pub trait CoordTrait {
+    /// The coordinate type of this geometry
+    type T: CoordNum;
+
+    /// Dimensions of the coordinate tuple
+    fn dim(&self) -> Dimensions;
+
+    /// Access the n'th (0-based) element of the CoordinateTuple.
+    /// Returns NaN if `n >= DIMENSION`.
```

Comment:
```suggestion
    /// Returns `None` if `n >= DIMENSION`.
```

Is this a typo or should it return `NaN`?